build-logic/src/main/kotlin/publish-library-convention.gradle.kts

```diff
@@ -0,0 +1,50 @@
+@file:Suppress("UnstableApiUsage")
+
+plugins {
+    id("com.vanniktech.maven.publish")
+}
+
+val versionSuffix = when (System.getenv("SNAPSHOT")) {
+    null -> "-TEST"
+    "true" -> "-SNAPSHOT"
+    else -> ""
+}
+project.version = project.version.toString() + versionSuffix
+
+mavenPublishing {
+    publishToMavenCentral(com.vanniktech.maven.publish.SonatypeHost.S01)
+
+    coordinates(
+        groupId = project.group.toString(),
+        artifactId = project.ext["artifactId"].toString(),
```

Comment:
[Improvement] no need to use ext map. ext["artifactId"] = "mvi-core" in lib project can be removed. This is actually project.name which is set to projects folder if not set directly.

Reply:
thx, will check it